rct229/rulesets/ashrae9012019/section19/section19rule2.py

```diff
@@ -0,0 +1,207 @@
+                    "chilled_water_loop",
+                    "cooling_system",
+                    "chilled_water_loop",
+                )
+                CHW_fluid_loop_list.append(chilled_water_loop_b)
+                if chilled_water_loop_b in chiller_cooling_loop_b:
+                    chiller_b = find_all_with_field_value(
+                        "$.chillers[*]",
+                        "cooling_loop",
+                        getattr_(
+                            hvac_b, "HVAC", "cooling_system", "chilled_water_loop"
+                        ),
+                        rmi_b,
+                    )
+                    if chiller_b.get("condensing_loop"):
+                        CW_fluid_loop_list.append(chiller_b["condensing_loop"])
+                else:
+                    primary_loop_b = find_exactly_one_child_loop(
+                        rmi_b, chilled_water_loop_b
+                    )
+                    CHW_fluid_loop_list.append(primary_loop_b["id"])
+
+        for CHW_child_loop in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].cooling_system.chilled_water_loop",
+            rmi_b,
+        ):
+            CHW_fluid_loop_list.append(CHW_child_loop)
+
+        # extend `HW_fluid_loop_list` with terminals
+        HW_fluid_loop_list.extend(
+            [
+                terminal["heating_from_loop"]
+                for terminal in find_all(
+                    "$.buildings[*].building_segments[*].zones[*].terminals[*]", rmi_b
+                )
+                if (
+                    terminal.get("heating_source") == HEATING_SOURCE.HOT_WATER
+                    and are_all_terminal_heating_loops_attached_to_boiler(
+                        rmi_b, [terminal["id"]]
+                    )
+                )
+            ]
+        )
+
+        return {
+            "HW_fluid_loop_list": list(set(HW_fluid_loop_list)),
+            "CHW_fluid_loop_list": list(set(CHW_fluid_loop_list)),
+            "CW_fluid_loop_list": list(set(CW_fluid_loop_list)),
+        }
+
+    class FluidLoopRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section19Rule2.FluidLoopRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["type"],
+                },
+            )
+
+        def is_applicable(self, context, data=None):
+            fluid_loop_b = context.baseline
+            fluid_loop_id_b = fluid_loop_b["id"]
+
+            HW_fluid_loop_list = data["HW_fluid_loop_list"]
+            CHW_fluid_loop_list = data["CHW_fluid_loop_list"]
+            CW_fluid_loop_list = data["CW_fluid_loop_list"]
+
+            return (
+                fluid_loop_id_b in HW_fluid_loop_list
+                or fluid_loop_id_b in CHW_fluid_loop_list
+                or CW_fluid_loop_list in CW_fluid_loop_list
```

Comment:
I think you mean `or fluid_loop_id_b in CW_fluid_loop_list`.

An alternate approach would be 
```
return fluid_loop_id_b in (HW_fluid_loop_list + CHW_fluid_loop_list + CW_fluid_loop_list)
```
This works because `+` concatenates the lists.

Reply:
Addressed.